src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,42 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- set-via-env? [setting]
+  (contains? env/env (setting-env-map-name setting)))
+
+(defmulti ^:private validate-setting-format :type)
+
+;; Treat setting as valid unless a validator has been defined for its type
+(defmethod validate-setting-format :default [_] nil)
+
+(defmethod validate-setting-format :json [{:keys [name]}]
+  (try
+    ;; attempt to parse the string to trigger any related exception
+    (doall (json/parse-string (get-value-of-type :string name)))
+    ;; there was no issue parsing the setting, so return a falsey value
+    nil
+    (catch JsonParseException e e)
+    ;; be conservative about what is returned, to avoid accidentally exposing secrets in the value, so do
+    (catch Exception e (ex-info "Unable to validate setting" {:exception-type (type e)}))))
+
+(defn- validate-setting [setting]
+  (when-let [error (validate-setting-format setting)]
+    (assoc (select-keys setting [:name :type])
+      :parse-error error
+      :env-var? (set-via-env? setting))))
+
+(defn validate-settings-formatting!
+  "Check whether there are any issues with the format of application settings, e.g. an invalid JSON string.
+
+   Note that this will only check settings whose [[defsetting]] forms have already been evaluated."
+  []
+  (doseq [invalid-setting (keep validate-setting (vals @registered-settings))]
+    (if (:env-var? invalid-setting)
+      (throw (ex-info (trs "Invalid {0} configuration for setting: {1}"
+                           #_:clj-kondo/ignore (str/upper-case (name (:type invalid-setting)))
+                           (name (:name invalid-setting)))
+                      (dissoc invalid-setting :parse-error)
+                      (:cause (:parse-error invalid-setting))))
```

Comment:
I see you're getting the `:cause` from the `:parse-error` value in both cases. What do you think of simplifying things and returning the thing we want displayed to the user in the `validate-setting-format` function?

Reply:
I went this route as it will keep things DRY as we add validation for more types, and also kept the type signature of the multimethod simpler. The latter is great as it simplifies the docstring I need to write 😆